src/components/RecipeLinks.astro

```diff
@@ -1,39 +1,56 @@
 ---
-import { getEntryBySlug } from 'astro:content';
-import { removeLeadingSlash, removeTrailingSlash } from '~/util';
+import { recipePages } from '~/content';
+import { getLanguageFromURL, stripLangFromSlug } from '~/util';
 
 export interface Props {
 	slugs: string[];
 }
 
-const slugs = Astro.props.slugs.map((slug) => removeLeadingSlash(removeTrailingSlash(slug)));
+const lang = getLanguageFromURL(Astro.url.pathname);
+const fallbackSlug = (slug: string) =>
+	'en/' + stripLangFromSlug(slug as Parameters<typeof stripLangFromSlug>[0]);
 
-const entries = await Promise.all(
-	slugs.map(async (slug, i) => {
-		const entry = await getEntryBySlug('docs', slug);
-		if (!entry) {
-			throw new Error(`Could not find entry for slug "${Astro.props.slugs[i]}"`);
-		}
-		return entry;
-	})
-);
+const recipes = Astro.props.slugs.map((slug) => {
+	let isFallback = lang !== 'en' && slug.startsWith('en/');
+	let entry = recipePages.find((recipe) => recipe.slug === slug);
+	if (!entry) {
+		isFallback = true;
+		entry = recipePages.find((recipe) => recipe.slug === fallbackSlug(slug));
+	}
+	if (!entry) {
+		throw new Error(`Could not find entry for slug "${slug}"`);
+	}
+	return { entry, isFallback };
+});
 
-const isList = entries.length > 1;
+const isList = recipes.length > 1;
 const noun = isList ? 'recipes' : 'recipe';
+const firstRecipe = recipes[0];
+if (!firstRecipe) {
+	throw new Error('No slugs passed to `<RecipeLinks>` component. Make sure you pass at least one.');
+}
 ---
 
 <div class="root">
 	<div class="flex">
 		<img src="/houston_chef.webp" width="26" alt="" />
 		<strong>Related {noun}:</strong>
```

Comment:
Non-blocking, but I'm wondering if we can do something to allow translating these? I know singular/plural translation stuff can get tricky in some languages, though we could just use something like "Related recipe(s)" and allow translators to add the entire thing even if it's not perfect.

Reply:
Oh yeah! Good point. Let’s do it in a separate PR so we can at least get your translation PRs unblocked.

Simplest here is probably to have two translation keys, one for singular, one for plural — “Related recipe” and “Related recipes” in English — to accommodate whatever plurals look like in each language.